math/liblinear/files/patch-Makefile.diff

```diff
@@ -1,50 +1,39 @@
-*** Makefile.orig	Thu Apr 12 03:38:27 2012
---- Makefile	Sat Jun  9 21:12:44 2012
-***************
-*** 6,20 ****
-  OS = $(shell uname)
-  #LIBS = -lblas
-  
-! all: train predict
-  
-  lib: linear.o tron.o blas/blas.a
-  	if [ "$(OS)" = "Darwin" ]; then \
-! 		SHARED_LIB_FLAG="-dynamiclib -Wl,-install_name,liblinear.so.$(SHVER)"; \
-  	else \
-! 		SHARED_LIB_FLAG="-shared -Wl,-soname,liblinear.so.$(SHVER)"; \
-  	fi; \
-! 	$(CXX) $${SHARED_LIB_FLAG} linear.o tron.o blas/blas.a -o liblinear.so.$(SHVER)
-  
-  train: tron.o linear.o train.c blas/blas.a
-  	$(CXX) $(CFLAGS) -o train train.c tron.o linear.o $(LIBS)
---- 6,24 ----
-  OS = $(shell uname)
-  #LIBS = -lblas
-  
-! PREFIX ?= /usr/local
-! 
-! all: train predict lib
-  
-  lib: linear.o tron.o blas/blas.a
-  	if [ "$(OS)" = "Darwin" ]; then \
-! 		LIBEXT=".dylib"; \
-! 		SHARED_LIB_FLAG="-dynamiclib -install_name $(PREFIX)/lib/liblinear$${LIBEXT}"; \
-  	else \
-! 		LIBEXT=".so.$(SHVER)"; \
-! 		SHARED_LIB_FLAG="-shared -Wl,-soname,liblinear$${LIBEXT}"; \
-  	fi; \
-! 	$(CXX) $${SHARED_LIB_FLAG} linear.o tron.o blas/blas.a -o liblinear$${LIBEXT}
-  
-  train: tron.o linear.o train.c blas/blas.a
-  	$(CXX) $(CFLAGS) -o train train.c tron.o linear.o $(LIBS)
-***************
-*** 34,37 ****
-  clean:
-  	make -C blas clean
-  	make -C matlab clean
-! 	rm -f *~ tron.o linear.o train predict liblinear.so.$(SHVER)
---- 38,41 ----
-  clean:
-  	make -C blas clean
-  	make -C matlab clean
-! 	rm -f *~ tron.o linear.o train predict liblinear.*
+--- Makefile.orig	2023-02-17 20:01:34.000000000 +0800
++++ Makefile	2023-05-15 18:46:27.000000000 +0800
+@@ -3,18 +3,22 @@
+ CFLAGS = -Wall -Wconversion -O3 -fPIC
+ LIBS = blas/blas.a
+ #LIBS = -lblas
+-SHVER = 5
++#SHVER = 5
++PREFIX ?= /opt/local
++
+ OS = $(shell uname)
+-ifeq ($(OS),Darwin)
+-	SHARED_LIB_FLAG = -dynamiclib -Wl,-install_name,liblinear.so.$(SHVER)
+-else
+-	SHARED_LIB_FLAG = -shared -Wl,-soname,liblinear.so.$(SHVER)
+-endif
+ 
+-all: train predict
++all: train predict lib
+ 
+ lib: linear.o newton.o blas/blas.a
+-	$(CXX) $(SHARED_LIB_FLAG) linear.o newton.o blas/blas.a -o liblinear.so.$(SHVER)
++	if [ "$(OS)" = "Darwin" ]; then \
++		LIBEXT=".dylib"; \
++		SHARED_LIB_FLAG="-dynamiclib -install_name $(PREFIX)/lib/liblinear$${LIBEXT}"; \
++	else \
++		LIBEXT=".so"; \
++		SHARED_LIB_FLAG="-shared -Wl,-soname,liblinear$${LIBEXT}"; \
++	fi; \
```

Comment:
Why have you removed the major version number from the library's install name? Assuming that the developers assign new major version numbers using the correct criteria, removing it here will lead to problems (programs that use the library crashing or not working properly) next time they increase it if we don't revbump all ports that link with the library at that time. (Granted, I don't see any such ports right now, but there could be in the future, and this also applies to any software outside of MacPorts that the user might have that links with the library.) Having the library filename change, via the change in major version number, makes it clear that a rebuild is needed so I'm very skeptical of you removing it.

Before your changes, the patch removed the major version number in the library filename on Darwin and kept it on all others. After your changes, you've removed it for all OS types. It doesn't seem like this change would be suitable for submission to the developers. I would instead keep it for all OS types.

Reply:
@ryandesign Well, it should either be kept for all OS or removed for all OS. As far as I can see, there is no reason to specifically remove it for Darwin. However in Macports we have only one version of the port and no plans to have multiple `liblinear` in parallel, I believe, so adding a number to dylib name is unnecessary.